Sourcery/Generating/Generator.swift

```diff
@@ -106,10 +99,10 @@ enum Generator {
             }
 
             baseType.based.keys.forEach {  type.based[$0] = $0 }
-            baseType.inherits.keys.forEach {  type.inherits[$0] = $0 }
             baseType.implements.keys.forEach {  type.implements[$0] = $0 }
 
-            if baseType.isClass {
+            if let type = type as? Class, let baseType = baseType as? Class {
+                baseType.inherits.keys.forEach {  type.inherits[$0] = $0 }
```

Comment:
I wouldn't be putting this expression here, it's impossible for swift to allow wrong logic here, that's why I put inherits and implements above this if statement